JobServiceContext.doDequeueWork(JobServiceContext.java:371)

## 💥 Crash Report
This seems to be connected to re-enabling android background sync.

### 👥 Impact
Affected around 0.1% of our users
So far this happened only on **Pixel devices** running **Android 10**. 

### 🔗 Context
Maybe this could be related to enabling background sync - https://github.com/toggl/mobileapp/issues/4298

### 🛰 Link
https://appcenter.ms/orgs/Toggl/apps/Toggl-Android/crashes/errors/1263678719u/overview

### 📝  Exception & Stack Trace
`com.microsoft.appcenter.crashes.Crashes.saveUncaughtException
Crashes.java, line 970`
```cs
java.lang.RuntimeException: An error occurred while executing doInBackground()
  at android.os.AsyncTask$4.done(AsyncTask.java:399)
  at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
  at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
  at java.util.concurrent.FutureTask.run(FutureTask.java:271)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
  at java.lang.Thread.run(Thread.java:919)
  Caused by: java.lang.SecurityException: Caller no longer running, last stopped +5s170ms because: timed out while starting
    at android.os.Parcel.createException(Parcel.java:2071)
    at android.os.Parcel.readException(Parcel.java:2039)
    at android.os.Parcel.readException(Parcel.java:1987)
    at android.app.job.IJobCallback$Stub$Proxy.dequeueWork(IJobCallback.java:292)
    at android.app.job.JobParameters.dequeueWork(JobParameters.java:248)
    at android.support.v4.app.JobIntentService$JobServiceEngineImpl.dequeueWork(JobIntentService.java:315)
    at android.support.v4.app.JobIntentService.dequeueWork(JobIntentService.java:640)
    at android.support.v4.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:390)
    at android.support.v4.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
    at android.os.AsyncTask$3.call(AsyncTask.java:378)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    at java.lang.Thread.run(Thread.java:919)
    Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.job.JobServiceContext.assertCallerLocked(JobServiceContext.java:493)
	at com.android.server.job.JobServiceContext.doDequeueWork(JobServiceContext.java:371)
	at com.android.server.job.JobServiceContext$JobCallback.dequeueWork(JobServiceContext.java:160)
	at android.app.job.IJobCallback$Stub.onTransact(IJobCallback.java:169)
	at android.os.Binder.execTransactInternal(Binder.java:1021)

```

--- a/Toggl.Droid/Services/BackgroundSyncJobSchedulerService.cs
+++ b/Toggl.Droid/Services/BackgroundSyncJobSchedulerService.cs
@@ -2,6 +2,7 @@
 using Android.App.Job;
 using System;
 using System.Reactive.Linq;
+using Toggl.Droid.Helper;
 using static Toggl.Shared.Extensions.CommonFunctions;
 
 namespace Toggl.Droid.Services
@@ -17,6 +18,14 @@ namespace Toggl.Droid.Services
 
         public override bool OnStartJob(JobParameters @params)
         {
+            // Background sync is temporary disabled due to a crash on Android 10 that is hard to reproduce
+            // Calling JobFinished and returning early here stops the background job from running
+            if (QApis.AreAvailable)
+            {
+                JobFinished(@params, false);
+                return true;
+            }
+
             AndroidDependencyContainer.EnsureInitialized(ApplicationContext);
             var dependencyContainer = AndroidDependencyContainer.Instance;
             if (!dependencyContainer.UserAccessManager.CheckIfLoggedIn())

--- a/Toggl.Droid/Services/BackgroundSyncJobSchedulerService.cs
+++ b/Toggl.Droid/Services/BackgroundSyncJobSchedulerService.cs
@@ -23,7 +23,7 @@ namespace Toggl.Droid.Services
             if (QApis.AreAvailable)
             {
                 JobFinished(@params, false);
-                return true;
+                return false;
             }
 
             AndroidDependencyContainer.EnsureInitialized(ApplicationContext);

--- a/Toggl.Droid/Services/BackgroundSyncServiceAndroid.cs
+++ b/Toggl.Droid/Services/BackgroundSyncServiceAndroid.cs
@@ -3,6 +3,7 @@ using Android.App.Job;
 using Android.Content;
 using Toggl.Core.Services;
 using Toggl.Droid.Extensions;
+using Toggl.Droid.Helper;
 
 namespace Toggl.Droid.Services
 {
@@ -10,6 +11,13 @@ namespace Toggl.Droid.Services
     {
         public override void EnableBackgroundSync()
         {
+            // Background sync is temporary disabled for android 10 due to a crash that is hard to reproduce
+            if (QApis.AreAvailable)
+            {
+                DisableBackgroundSync();
+                return;
+            }
+
             var context = Application.Context;
             var jobScheduler = (JobScheduler)context.GetSystemService(Context.JobSchedulerService);
             var periodicity = (long)MinimumBackgroundFetchInterval.TotalMilliseconds;

